### PR TITLE
Ascanrules: Fix .htaccess FPs

### PR DESCRIPTION
**Pull request by psiinon**
Fixes https://github.com/zaproxy/zaproxy/issues/7632

## Overview
Added more checks for valid .htaccess files to reduce false positives

## Related Issues
https://github.com/zaproxy/zaproxy/issues/7632

## Checklist
- [ ] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/HtAccessScanRule.java [hunk 1]
```diff
@@ -67,20 +77,41 @@ public boolean isFalsePositive(HttpMessage msg) {
...
         }
-
-        return false;
+        // Check for common htaccess directives in the first 10k
+        String body = msg.getResponseBody().toString();
+        if (body.length() > TEN_K) {
+            body = body.substring(0, TEN_K);
+        }
+        final String body10k = body.toLowerCase();
+        if (!COMMON_DIRECTIVES.stream().anyMatch(d -> body10k.contains(d))) {
```
Comment (kingthorin) [inline addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/HtAccessScanRule.java:96]: ignore case?

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/HtAccessScanRule.java [hunk 2]
```diff
@@ -39,6 +40,11 @@ public class HtAccessScanRule extends AbstractAppFilePlugin implements CommonAct
                     CommonAlertTag.OWASP_2017_A06_SEC_MISCONFIG,
                     CommonAlertTag.WSTG_V42_CONF_05_ENUMERATE_INFRASTRUCTURE);
 
+    private static final int TEN_K = 10 * 1024;
+
+    private static final List<String> COMMON_DIRECTIVES =
+            List.of("allow", "deny", "files", "limit", "options", "order", "redirect", "require");
```
[the remark this answers is not in the document]
Comment (psiinon) [reply addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/HtAccessScanRule.java:46]: Heres the full list https://httpd.apache.org/docs/trunk/mod/directives.html
but I dint want to include all of those! Where do we draw the line?

File: addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/HtAccessScanRuleUnitTest.java [hunk 3]
```diff
@@ -98,6 +100,26 @@ void shouldReturnExpectedMappings() {
...
+        // Given
+        TechSet techSet = new TechSet(Tech.C, Tech.Apache, Tech.ASP);
+        // When
+        boolean targets = rule.targets(techSet);
+        // Then
+        assertThat(targets, is(equalTo(true)));
+    }
+
+    @Test
+    void shouldNotTargetIfNoApache() throws Exception {
```
[the remark this answers is not in the document]
Comment (psiinon) [reply addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/HtAccessScanRuleUnitTest.java:114]: I dont think so https://stackoverflow.com/a/10528694/1509834

**Comment by psiinon** [comment]
Updated.

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/HtAccessScanRule.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/HtAccessScanRule.java]: Okay, fair. I just figured those mentioned in one of the top few example articles on google would be a good move but 🤷‍♂️ You're right it's hard to find the balance,

File: addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/HtAccessScanRuleUnitTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (kingthorin) [reply addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/HtAccessScanRuleUnitTest.java]: Ok

**Review by kingthorin — APPROVED** [review]

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**